packages/aws-sdk-signers/src/aws_sdk_signers/signers.py

```diff
@@ -789,6 +800,113 @@ async def _compute_payload_hash(
         return checksum.hexdigest()
 
 
+class AsyncEventSigner:
+    def __init__(
+        self,
+        *,
+        signing_properties: SigV4SigningProperties,
+        identity: AWSCredentialIdentity,
+        initial_signature: bytes,
+    ):
+        self._signing_properties = signing_properties
+        self._identity = identity
+        self._prior_signature = initial_signature
+        self._signing_lock = asyncio.Lock()
+
+    async def sign_event(
+        self,
+        *,
+        event_message: "EventMessage",
+        event_encoder_cls: type["EventHeaderEncoder"],
+    ) -> "EventMessage":
+        async with self._signing_lock:
+            # Copy and prepopulate any missing values in the
+            # signing properties.
+            new_signing_properties = SigV4SigningProperties(  # type: ignore
+                **self._signing_properties
+            )
+            # TODO: If date is in properties, parse a datetime from it.
```

Comment:
We might want to fix this now since the other signing methods do this and I think we'd probably want it for testing.  I think we should be able to use `_normalize_signing_properties` to create the new_signing_properties and then parse date_obj from it.